packages/api-server/api_server/models/door_state.py

```diff
@@ -0,0 +1,33 @@
+from rmf_door_msgs.msg import DoorMode as RmfDoorMode
+from rmf_door_msgs.msg import DoorState as RmfDoorState
+from tortoise import fields
+from tortoise.models import Model
+
+from ..ros_time import py_to_ros_time, ros_to_py_datetime
+from .door_mode import DoorModeEnum
+
+
+class DoorState(Model):
+    # TODO: foreign key with Door
```

Comment:
Mostly yea, but that is mostly an experimental, I didn't end up using the mixins because turns out every single field in `door_states` needs some model relation, it remains to be seen how useful the mixins will be for other models. Maybe it is possible to generate the relations base on the message but I don't know how to do it.

Reply:
Got it. 

> Maybe it is possible to generate the relations base on the message but I don't know how to do it.

You mean something like can create models and relations by introspecting existing messages (there is a tool for that but it only works with SQL tables =/ ). Just thinking out loud here, if we do that we also need to make a new "migration" every time a msg format changes, before receiving the message.

Reply:
> there is a tool for that but it only works with SQL tables =/ 

Thats very interesting, perhaps we can find a way to make it generate tortoise-orm models as well?

> Just thinking out loud here, if we do that we also need to make a new "migration" every time a msg format changes, before receiving the message.

I think this also brings up a point in regards to updating message, the impact and migration strategy should be carefully analyzed in their respective PR, in some cases, it might not be possible for `rmf-web` to perform the migration. For example, if the new messages contains a *required* field, we can't migrate by creating a new nullable column, that would violate the schema, so the message PR would need to describe how to fill in the new fields for existing records, or simply acknowledge that updating to the new version will be a breaking change, not just in terms of API/ABI, but also all database will be invalidated and need to restart from an empty state.